shell/browser/native_window_mac.mm

```diff
@@ -1731,22 +1661,42 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   }
 }
 
-void NativeWindowMac::OverrideNSWindowContentView() {
-  // When using `views::Widget` to hold WebContents, Chromium would use
-  // `BridgedContentView` as content view, which does not support draggable
-  // regions. In order to make draggable regions work, we have to replace the
-  // content view with a simple NSView.
-  if (has_frame()) {
-    container_view_.reset(
-        [[ElectronAdaptedContentView alloc] initWithShell:this]);
-  } else {
-    container_view_.reset([[FullSizeContentView alloc] init]);
-    [container_view_ setFrame:[[[window_ contentView] superview] bounds]];
+class NativeAppWindowFrameViewMac : public views::NativeFrameViewMac {
+ public:
+  NativeAppWindowFrameViewMac(views::Widget* frame, NativeWindowMac* window)
+      : views::NativeFrameViewMac(frame), native_window_(window) {}
+
+  NativeAppWindowFrameViewMac(const NativeAppWindowFrameViewMac&) = delete;
+  NativeAppWindowFrameViewMac& operator=(const NativeAppWindowFrameViewMac&) =
+      delete;
+
+  ~NativeAppWindowFrameViewMac() override = default;
+
+  // NonClientFrameView:
+  int NonClientHitTest(const gfx::Point& point) override {
+    if (!bounds().Contains(point))
+      return HTNOWHERE;
+
+    if (GetWidget()->IsFullscreen())
+      return HTCLIENT;
+
+    // Check for possible draggable region in the client area for the frameless
+    // window.
+    SkRegion* draggable_region = native_window_->draggable_region();
+    if (draggable_region && draggable_region->contains(point.x(), point.y()))
+      return HTCAPTION;
+
+    return HTCLIENT;
   }
-  [container_view_
-      setAutoresizingMask:NSViewWidthSizable | NSViewHeightSizable];
-  [window_ setContentView:container_view_];
-  AddContentViewLayers();
+
+ private:
+  // Weak.
+  raw_ptr<NativeWindowMac> native_window_;
```

Comment:
More of a style suggestion than a problem with the existing code, but as `native_window_` does not change for the life cycle of this object, this could possibly be

```suggestion
  raw_ptr<NativeWindowMac> const native_window_;
```